Update README.md (#3)
etecs-ru/typedjson: README.md

This is a code generator for Go that alleviates JSON marshaling/unmarshaling unrelated structs in typed fashion.

Imagine, that you send or receive JSON object with some key `config`. 
The value of this field can correspond to two structs of your Go program: `FooConfig` and `BarConfig`.
So, the field `Config` in your struct must be able to hold a value of two possible types.
In this case you basically have next options: 

1. You can declare field `Config` as `interface{}` and somehow determine what type you should expect, assign object of this type to `Config` 
and then unmarshal object.
1. You can unmarshal field `Config` separately.
1. You can implement custom `MarshalJSON`/`UnmarshalJSON` for third type that automatically will handle this cases.

This package provides means to generate all boilerplate code for third case.

## Usage

```sh
typedjson [OPTION] NAME...
```

Options:

* `-interface` string

	Name of the interface that encompass all types.

* `-output` string

	Output path where generated code should be saved.

* `-package` string

	Package name in generated file (default to GOPACKAGE).

* `-typed` string

> 	Name of struct that will used for typed interface (default to %%interface%%Typed.

Author: ```suggestion
	The name of the struct that will be used for typed interface (default to `{{interface}}{{Typed}}`).
```

This is a code generator for Go that alleviates JSON marshaling/unmarshaling unrelated structs in typed fashion.

> Imagine, that you send or receive JSON object with some key `config`. 

Reviewer: ```suggestion
Imagine, that you send or receive a JSON object with some key `config`. 
```

The value of this field can correspond to two structs of your Go program: `FooConfig` and `BarConfig`.

> So, the field `Config` in your struct must be able to hold value of two possible types.

Reviewer: ```suggestion
So, the field `Config` in your struct must be able to hold a value of two possible types.
```

> In this case you basically have next options: 

Reviewer: ```suggestion
In this case, you have the following options:
```

> 1. You can declare field `Config` as `interface{}` and somehow determine what type you should expect, assign object of this type to `Config` 

Reviewer: ```suggestion
1. You can declare field `Config` as `interface{}` and somehow determine what type you should expect, assign an object of this type to `Config` 
```

and then unmarshal object.
1. You can unmarshal field `Config` separately.

> 1. You can implement custom `MarshalJSON`/`UnmarshalJSON` for third type that automatically will handle this cases.

Reviewer: ```suggestion
1. You can implement custom `MarshalJSON`/`UnmarshalJSON` for the third type that automatically will handle these cases.
```

> This package provides means to generate all boilerplate code for third case.

Reviewer: ```suggestion
This package provides means to generate all boilerplate code for the third case.
```

## Usage

```sh
typedjson [OPTION] NAME...
```

Options:

* `-interface` string

	Name of the interface that encompass all types.

* `-output` string

	Output path where generated code should be saved.

* `-package` string

	Package name in generated file (default to GOPACKAGE).

* `-typed` string

> 	Name of struct that will used for typed interface (default to %%interface%%Typed.

Reviewer: ```suggestion
	Name of the struct that will be used for typed interface (default to `%%interface%%Typed`).
```

> Each name in position argument should be name of struct. 

Reviewer: ```suggestion
Each name in position argument should be the name of the struct. 
```

> You can set alias for struct name like this: `foo=*FooConfig`.

Reviewer: ```suggestion
You can set an alias for struct name like this: `foo=*FooConfig`.
```

## Example

For example, you have two structs:

```go
type FooConfig struct {
	Foo int
}

type BarConfig struct {
	Bar string
}
```

> Then you must declare interface that will hold either of these structs.

Reviewer: ```suggestion
Then you must declare an interface that will hold either of these structs.
```

> The interface must have method `TypedJSON` with special signature. 

Reviewer: ```suggestion
The interface must have the method `TypedJSON` with a special signature. 
```

> This method help compiler to work with types.

Reviewer: ```suggestion
This method will advise compiler to work with types.
```
